### ISSUE TITLE
[Ch 17][Notebook][Minor] Loss 노트 「숫자로 감 잡기」 평균 손실 0.387 — 같은 줄에 적힌 식을 계산하면 0.357

### ISSUE THREAD

**Issue by jwrim9305** [yoon-gu/neuqes-101#275]
## 대상

- **챕터**: Ch 17 — 한국어 BERT Multi-label (KLUE-YNAT 합성)
- **산출물**: Notebook `17_ko_multilabel.ipynb` c3 · `pages/17-ko_multilabel.md:76` · `book/chapters/ch17_ko_multilabel.tex:198`
- **기준 커밋**: `5629f80` (master)
- 정적 검토 + 손계산 검산. Colab 재실행은 별건으로 수행(아래 「추가 확인」)

## 실제 결과

📐 Loss 노트의 「숫자로 감 잡기」가 **계산식을 그대로 써 놓고 답을 틀렸습니다.**

```
평균 loss = (0.048 + 0.474 + 0.127 + 1.704 + 0.048 × 3) / 7 ≈ 0.387
```

괄호 안을 더하면 `2.497` 이고, 7로 나누면 **`0.3567`** 입니다.

```
0.048 + 0.474 + 0.127 + 1.704 = 2.353
0.048 × 3                     = 0.144
합                            = 2.497
2.497 / 7                     = 0.3567   ← 책은 0.387
```

**표의 각 행은 전부 정확합니다.** 검산했습니다.

| 라벨 | $y_k$ | $z_k$ | 표의 $\sigma$ | 실제 $\sigma$ | 표의 손실 | 실제 손실 |
|---|---|---|---|---|---|---|
| 경제 | 1 | 3.0 | 0.953 | 0.9526 | 0.048 | 0.0486 ✅ |
| 스포츠 | 1 | 0.5 | 0.622 | 0.6225 | 0.474 | 0.4741 ✅ |
| IT과학 | 0 | −2.0 | 0.119 | 0.1192 | 0.127 | 0.1269 ✅ |
| 사회 | 0 | 1.5 | 0.818 | 0.8176 | 1.704 | 1.7014 ✅ |
| 나머지 3개 | 0 | −3.0 | 0.047 | 0.0474 | 각 0.048 | 0.0486 ✅ |

**틀린 것은 마지막 평균 한 개**입니다.

## 왜 이게 문제인가

이 표는 학습자가 **손으로 검산해 보라고 둔 교육용 예시**입니다. 더할 값까지 본문에 적어 두었으니, 계산해 보는 독자는 **반드시 다른 답을 얻습니다.** 각 행이 맞기 때문에 독자는 자기 계산을 의심하게 됩니다.

`0.387` 과 `0.357` 의 차이는 약 8%입니다. BCE per-label 의 직관("자신 있게 틀린 라벨 하나가 평균을 끌어올린다")을 전달하는 데 쓰이는 숫자라, 값이 틀리면 그 감각도 어긋납니다.

## 같은 계열의 선례

| 이슈 | 챕터 | 내용 |
|---|---|---|
| #154 | Ch 16 | Loss 노트 K=7 softmax 표 — 4행 중 3행 오류 |
| #161 | Ch 12 | Loss 노트 K=5 softmax 표 — 4행 중 3행 오류 |

**같은 자리의 같은 성격**이되 양상이 다릅니다. Ch 12·16 은 *행의 값* 이 틀렸고, Ch 17 은 *행은 맞고 합계* 가 틀렸습니다. Ch 13 의 같은 표(평균 `0.665`)는 검산해 보니 정확합니다.

## 기대 결과

숫자 하나만 고칩니다.

```diff
- 평균 loss = $(0.048 + 0.474 + 0.127 + 1.704 + 0.048 	imes 3) / 7 pprox 0.387$.
+ 평균 loss = $(0.048 + 0.474 + 0.127 + 1.704 + 0.048 	imes 3) / 7 pprox 0.357$.
```

## 재현 절차

```bash
python3 -c "
print((0.048 + 0.474 + 0.127 + 1.704 + 0.048*3) / 7)"     # 0.3567
```

```python
import math
sig = lambda z: 1/(1+math.exp(-z))
rows = [(1,3.0), (1,0.5), (0,-2.0), (0,1.5)] + [(0,-3.0)]*3
tot = sum(-math.log(sig(z)) if y else -math.log(1-sig(z)) for y,z in rows)
print(tot, tot/7)      # 2.4968  0.3567
```

## 추가 확인

같은 챕터를 Colab T4 에서 재실행해 대조했습니다(2026-09-13, `source_sha256` 일치). 이 표는 **마크다운 고정 예시**라 실행 결과와 무관하며, 재실행으로 바뀌지 않습니다.

## 전파 범위

```bash
grep -rn "0\.387" 17_ko_multilabel/ pages/17-*.md book/chapters/ch17_ko_multilabel.tex
```

| 축 | 상태 |
|---|---|
| 📓 Notebook | `17_ko_multilabel.ipynb` c3 **1곳** |
| 📄 WikiDocs | `pages/17-ko_multilabel.md:76` **1곳** |
| 📕 EPUB | `ch17_ko_multilabel.tex:198` **1곳** |

**재실행 불필요 · 재변환 불필요(마크다운이라 재변환으로 전파됨) · tex 는 재생성 또는 손수정.**

## 심각도

**Minor** 로 제안합니다.

선례인 Ch 12 #161 · Ch 16 #154 는 Major 였지만 **양상이 다릅니다.** 그쪽은 $\sigma$ 값 자체가 틀려(4행 중 3행) 독자가 **틀린 수학을 배우게** 되는 상태였습니다.

이 건은 **수식도 각 행의 값도 전부 정확**하고, 마지막 덧셈 결과 하나만 8% 어긋납니다. 전달하려는 직관(「자신 있게 틀린 라벨 하나가 평균을 끌어올린다」)도 0.357 에서 그대로 성립합니다.

다만 **독자가 검산하도록 설계된 자리**라 교정은 필요합니다. 등급 상향 의견이 있으면 맞추겠습니다.

## 담당

- 수정 담당자: 미정
- 재검수자: 등록자 (@jwrim9305)
- 연결: 챕터 검수 이슈 #41 / 계열 #154 (Ch 16) · #161 (Ch 12)
- **Ch 18 확인 권장** — 같은 형식의 Loss 노트가 있으면 합계도 검산해 보는 편이 좋겠습니다.

**Comment by jwrim9305** [comment]
## 수정 완료 (@jwrim9305), 2026-09-13

브랜치 `fix/ch17-review-2026-09-b1` @ `f523305`.

```diff
- 평균 loss = $(0.048 + 0.474 + 0.127 + 1.704 + 0.048 \times 3) / 7 \approx 0.387$.
+ 평균 loss = $(0.048 + 0.474 + 0.127 + 1.704 + 0.048 \times 3) / 7 \approx 0.357$.
```

표의 각 행은 건드리지 않았습니다 — σ 값과 라벨별 손실은 원래 정확했고, **마지막 평균 숫자 하나만** 바꿨습니다.

| 축 | 위치 | 재실행 |
|---|---|---|
| 📓 Notebook | `17_ko_multilabel.ipynb` c3 | **불필요** (마크다운 고정 예시) |
| 📄 WikiDocs | `pages/17-ko_multilabel.md` | **불필요** |
| 📕 EPUB | `book/chapters/ch17_ko_multilabel.tex` 손수정 | **불필요** · 재빌드 시 반영 |

`0.387` 잔존 grep **0건**. 린터 `check_wikidocs_md.py` 위반 0건.

> `\approx` 기호는 그대로 두고 값만 교체했습니다. 실측이 아니라 **고정 계산값**이라 재실행·재생성으로 바뀌지 않습니다 — 「약」 표기로 뭉개지 않고 검산 가능한 자리수를 유지했습니다.

**Comment by jwrim9305** [comment]
## 병합 확인 — 종결합니다 (@jwrim9305), 2026-09-13

**PR #278** 이 `2655899` 로 master 에 병합됐습니다. 등록·수정이 모두 저라 병합 후 master 에서 직접 확인했습니다.

```
평균 loss = (0.048 + 0.474 + 0.127 + 1.704 + 0.048 × 3) / 7 ≈ 0.357
```

| 축 | 확인 | 재실행 |
|---|---|---|
| 📓 Notebook | c3 📐 Loss 노트 | 불필요 |
| 📄 WikiDocs | `pages/17-ko_multilabel.md` | 불필요 |
| 📕 EPUB | `ch021.xhtml` MathML 에 `0.357` | 불필요 |

`0.387` 잔존 **0건** (노트북·pages·tex·EPUB). 표의 각 행은 손대지 않았습니다 — 원래 정확했습니다.

> tex 재생성 뒤에도 유지됐습니다. 노트북에서 고쳤기 때문에 손수정과 달리 재생성에 안전합니다.
